transport/transport.go

```diff
@@ -341,8 +341,10 @@ func NewServerTransport(protocol string, conn net.Conn, maxStreams uint32, authI
 type ConnectOptions struct {
 	// UserAgent is the application user agent.
 	UserAgent string
-	// Dialer specifies how to dial a network address.
-	Dialer func(string, time.Duration) (net.Conn, error)
+	// Cancel is closed to indicate that dialing should be cancelled.
+	Cancel chan struct{}
+	// dialer specifies how to dial a network address.
+	dialer func(string, time.Duration, <-chan struct{}) (net.Conn, error)
```

Comment:
It's left unexported to encourage users to use the `Dial` method rather
than accessing this directly. That method encapsulates usage of the default.

On Jul 26, 2016 15:50, "Qi Zhao" <notifications@github.com> wrote:

> In transport/transport.go
> <https://github.com/grpc/grpc-go/pull/751#discussion_r72323639>:
>
> > @@ -341,8 +341,10 @@ func NewServerTransport(protocol string, conn net.Conn, maxStreams uint32, authI
> >  type ConnectOptions struct {
> >  	// UserAgent is the application user agent.
> >  	UserAgent string
> > -	// Dialer specifies how to dial a network address.
> > -	Dialer func(string, time.Duration) (net.Conn, error)
> > +	// Cancel is closed to indicate that dialing should be cancelled.
> > +	Cancel chan struct{}
> > +	// dialer specifies how to dial a network address.
> > +	dialer func(string, time.Duration, <-chan struct{}) (net.Conn, error)
>
> Any reason this is not exported? All the other fields are exported. If it
> is exported, it seems we do not need ConnectOptions.WithDialer(...).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/grpc/grpc-go/pull/751/files/e80127070134096b644d5877643696a67efebf3f#r72323639>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABdsPIK_NVK188sAbESZnw_tyMCWArPGks5qZmUbgaJpZM4JFbEQ>
> .
>